### PR TITLE
Makes things better

### PR DESCRIPTION
**Pull request by bishoy301**
Replace inferior language with glorious GC'd Scala

### CODE REVIEW

**Comment by sremedios** [comment]
This looks pretty good imo

File: HelloWorld.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+object HelloWorld {
+   /* This is my first java program.  
+    * This will print 'Hello World' as the output
+    * This is an example of multi-line comments.
+    */
+   def main(args: Array[String]) {
+      // Prints Hello World
+      // This is also an example of single line comment.
+      var name = "sina";
```
Comment (starfys) [inline HelloWorld.scala]: this is incorrect. you are bishoy

File: HelloWorld.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+      println("Hello, world!" + name)
```
Comment (starfys) [inline HelloWorld.scala]: string concatenation and not quasiquotes? wtf

File: HelloWorld.scala [hunk 3]
```diff
@@ -0,0 +1,12 @@
+object HelloWorld {
```
Comment (Riuchando) [inline HelloWorld.scala:2]: doesn't have enough Big data, I would suggest https://spark.apache.org/ or https://flink.apache.org/

File: demo.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+object Demo {
+   def main(args: Array[String]) {
+      /*val pt = new Point(5,50);
+      pt.move(10,10);
+
+      val loc = new Location(1,2,3);
+      loc.move(5,5,5);
+   */
+      4 times println("sina");
```
Comment (starfys) [inline demo.scala]: scala is weird dude

File: HelloWorld.scala (hunk 1 above)
Comment (Riuchando) [inline HelloWorld.scala]: change to a name of someone who is not a scrub, like Steven Sheffey

File: build/build.sbt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+scalaVersion := "2.11.7"
+
+name := "Simple Scala Project"
+
+libraryDependencies += "org.apache.spark" % "spark-mllib_2.11" % "1.5.1"
```
Comment (starfys) [inline build/build.sbt]: why import libraries you aren't using

File: cl.scala [hunk 6]
```diff
@@ -0,0 +1,9 @@
+object Cl {
+    //closure
+   def main(args: Array[String]) {
+      println( "multiplier(1) value = " +  multiplier(1) )
```
Comment (starfys) [inline cl.scala]: tfw no quasiquotes

File: HelloWorld.scala (hunk 2 above)
Comment (Riuchando) [inline HelloWorld.scala]: use string formatting or superior Java string builder

**Review by starfys — CHANGES_REQUESTED** [review]

File: Run.scala [hunk 7]
```diff
@@ -0,0 +1,13 @@
+object Run {
+   implicit class IntTimes(x: Int) {
+      def times [A](f: =>A): Unit = {
+         def loop(current: Int): Unit =
+         
+         if(current > 0){
```
Comment (Riuchando) [inline Run.scala:8]: please use a scaliform formatter: https://github.com/sbt/sbt-scalariform

File: Run.scala [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+            f
+            loop(current - 1)
```
Comment (Riuchando) [inline Run.scala:10]: when possible use for loops or at least add the `@tailrec` annotation  
https://www.scala-lang.org/api/2.12.3/scala/annotation/tailrec.html

File: build/build.sbt (hunk 5 above)
Comment (Riuchando) [inline build/build.sbt]: please bump spark to version 2.3.0 1.5.1 uses rdds instead of dataframes

File: cl.scala [hunk 9]
```diff
@@ -0,0 +1,9 @@
+object Cl {
+    //closure
```
Comment (Riuchando) [inline cl.scala]: add a space after comment

File: cl.scala (hunk 6 above)
Comment (Riuchando) [inline cl.scala]: fix formatting

File: demo.scala [hunk 10]
```diff
@@ -0,0 +1,40 @@
+import java.io._
+import Run._
+
+class Point(xc: Int, yc: Int) {
+   var x: Int = xc
```
Comment (Riuchando) [inline demo.scala:5]: get these dirty mutable variables out of here

**Review by Riuchando — CHANGES_REQUESTED** [review]

File: Run.scala [hunk 11]
```diff
@@ -0,0 +1,13 @@
+object Run {
+   implicit class IntTimes(x: Int) {
+      def times [A](f: =>A): Unit = {
+         def loop(current: Int): Unit =
+         
+         if(current > 0){
+            f
```
Comment (Riuchando) [inline Run.scala:9]: remove unnecessary variable call

File: HelloWorld.scala (hunk 1 above)
Comment (sremedios) [reply HelloWorld.scala]: hmmm :thonking:

File: cl.scala (hunk 9 above)
Comment (sremedios) [reply cl.scala]: comment` `

**Review by Riuchando — CHANGES_REQUESTED** [review]

File: demo.scala (hunk 10 above)
Comment (bishoy301) [reply demo.scala:5]: Ackshually, sometimes it's better to have mutable variables

https://docs.scala-lang.org/overviews/collections/performance-characteristics.html

**Comment by bishoy301** [comment]
@Riuchando Please review changes

File: Run.scala (hunk 11 above)
Comment (bishoy301) [reply Run.scala:9]: No, that's wrong

Comment (Riuchando) [reply Run.scala:9]: https://www.youtube.com/watch?v=e53HfowZCdI

File: build.sbt [hunk 12]
```diff
@@ -0,0 +1,13 @@
+scalaVersion := "2.12.8"
+
+name := "Dank Scala Project"
+
+libraryDependencies += "org.apache.spark" %% "spark-sql" % "2.4.1"
```
Comment (Riuchando) [inline build.sbt:5]: 2.3.0, gotta have that spark on kube integration

File: Run.scala [hunk 13]
```diff
@@ -1,7 +1,9 @@
+import scala.annotation.tailrec
+
 object Run {
    implicit class IntTimes(x: Int) {
       def times [A](f: =>A): Unit = {
-         def loop(current: Int): Unit =
+         @tailrec def loop(current: Int): Unit =
```
Comment (Riuchando) [inline Run.scala:6]: please put the tailrec on the line before

Comment (Riuchando) [reply Run.scala:6]: also add a parens around the if else statement

File: Run.scala [hunk 14]
```diff
@@ -1,7 +1,9 @@
+import scala.annotation.tailrec
+
 object Run {
    implicit class IntTimes(x: Int) {
       def times [A](f: =>A): Unit = {
```
Comment (Riuchando) [inline Run.scala:5]: the scaliform should add a space before the A

File: HelloWorld.scala [hunk 15]
```diff
@@ -1,3 +1,4 @@
+import org.apache.spark.sql.DataFrame
```
Comment (Riuchando) [inline HelloWorld.scala:1]: please utilize big data to println Helloworld, look at https://www.programcreek.com/scala/org.apache.spark.SparkConf

try to add

```
val conf:SparkConf = new SparkConf().setAppName("Histogram").setMaster("local")
val sc: SparkContext = new SparkContext(conf)
```

then add http://www.scalatest.org/getting_started_with_fun_spec that adds the spark dataframe

File: HelloWorld.scala [hunk 16]
```diff
@@ -6,7 +7,10 @@ object HelloWorld {
    def main(args: Array[String]) {
       // Prints Hello World
       // This is also an example of single line comment.
-      var name = "sina";
-      println("Hello, world!" + name) 
+      var name = "Chihiro Fujisaki";
```
Comment (Riuchando) [inline HelloWorld.scala:10]: remove semi colons, they are optional

File: HelloWorld.scala [hunk 17]
```diff
@@ -6,7 +7,10 @@ object HelloWorld {
    def main(args: Array[String]) {
```
Comment (Riuchando) [inline HelloWorld.scala:7]: add a `: Unit` to assert that the return is a none type

**Review by Riuchando — CHANGES_REQUESTED** [review]

File: HelloWorld.scala (hunk 16 above)
Comment (starfys) [reply HelloWorld.scala:10]: also this should be a val

File: README.md [hunk 18]
```diff
@@ -0,0 +1,4 @@
+# Scala
+
+## about
+Basic helloworld scala scripts to demonstrate basic understaing of scala.
```
Comment (starfys) [inline README.md:4]: typos

File: Run.scala (hunk 14 above)
Comment (starfys) [reply Run.scala]: scalariform doesn't seem to do that. weird

File: build.sbt (hunk 12 above)
Comment (starfys) [reply build.sbt:5]: koo ber nay tayce

File: demo.scala [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+      z = z + dz
+      println ("Point x location : " + x);
+      println ("Point y location : " + y);
+      println ("Point z location : " + z);
+   }
+}
+
+object Demo {
+   def main(args: Array[String]) {
+      /*val pt = new Point(5,50);
```
Comment (sremedios) [inline demo.scala:32]: Please remove deprecated code.

File: demo.scala (hunk 10 above)
Comment (Riuchando) [reply demo.scala:5]: avoid mutable variables in parallelized environments, especially if not necessary

Comment (bishoy301) [reply demo.scala:5]: It doesn't compile

File: HelloWorld.scala (hunk 15 above)
Comment (starfys) [reply HelloWorld.scala:1]: Are you sure an RDD isn't better for this?

File: demo.scala [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+
+class Location( val xc: Int,  val yc: Int,
+   val zc :Int) extends Point(xc, yc){
+   var z: Int = zc
+
+   def move(dx: Int, dy: Int, dz: Int) {
+      x = x + dx
+      y = y + dy
+      z = z + dz
+      println ("Point x location : " + x);
```
Comment (starfys) [inline demo.scala]: use quasiquotes here

File: demo.scala [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+      println ("Point y location : " + y);
```
Comment (starfys) [inline demo.scala:25]: use quasiquotes or stringbuilder

File: demo.scala [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+      println ("Point z location : " + z);
```
Comment (starfys) [inline demo.scala:26]: use quasiquotes or stringbuilder

File: demo.scala (hunk 4 above)
Comment (starfys) [inline demo.scala:38]: replace name

File: cl.scala [hunk 23]
```diff
@@ -0,0 +1,9 @@
+object Cl {
+   // closure
+   def main(args: Array[String]) {
+      println("multiplier(1) value = " +  multiplier(1))
```
Comment (starfys) [inline cl.scala:4]: use quasiquotes or stringbuilder

File: cl.scala [hunk 24]
```diff
@@ -0,0 +1,9 @@
...
+      println("multiplier(2) value = " +  multiplier(2))
```
Comment (starfys) [inline cl.scala:5]: use quasiquotes or stringbuilder

File: cl.scala [hunk 25]
```diff
@@ -0,0 +1,9 @@
...
+   }
+   var factor = 3
```
Comment (starfys) [inline cl.scala:7]: change to val

File: demo.scala [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+import Run._
+
+class Point(xc: Int, yc: Int) {
+   var x: Int = xc
+   var y: Int = yc
+
+   def move(dx: Int, dy: Int) {
+      x = x + dx
+      y = y + dy
+      println ("Point x location : " + x);
```
Comment (starfys) [inline demo.scala:11]: use quasiquotes or stringbuilder

File: demo.scala [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+      println ("Point y location : " + y);
```
Comment (starfys) [inline demo.scala:12]: use quasiquotes or stringbuilder

File: demo.scala (hunk 20 above)
Comment (starfys) [inline demo.scala:24]: use quasiquotes or stringbuilder

File: demo.scala [hunk 28]
```diff
@@ -0,0 +1,40 @@
...
+   }
+}
+
+class Location( val xc: Int,  val yc: Int,
+   val zc :Int) extends Point(xc, yc){
+   var z: Int = zc
+
+   def move(dx: Int, dy: Int, dz: Int) {
+      x = x + dx
```
Comment (starfys) [inline demo.scala:21]: does scala not have `+=`?

File: demo.scala [hunk 29]
```diff
@@ -0,0 +1,40 @@
+import java.io._
+import Run._
+
+class Point(xc: Int, yc: Int) {
```
Comment (starfys) [inline demo.scala:4]: What is this class used for?

File: demo.scala [hunk 30]
```diff
@@ -0,0 +1,40 @@
...
+
+   def move(dx: Int, dy: Int) {
+      x = x + dx
+      y = y + dy
+      println ("Point x location : " + x);
+      println ("Point y location : " + y);
+   }
+}
+
+class Location( val xc: Int,  val yc: Int,
```
Comment (starfys) [inline demo.scala:16]: What is this class used for?

File: HelloWorld.scala (hunk 15 above)
Comment (Riuchando) [reply HelloWorld.scala:1]: it'll be a learning experience for Bishoy

**Comment by Riuchando** [comment]
This was a fun joke, unfortunately things can not get better.

**Pull request closed without merging**